decidim-admin/app/controllers/decidim/admin/admin_terms_controller.rb

```diff
@@ -9,7 +9,7 @@ def accept
         current_user.admin_terms_accepted_at = Time.current
         if current_user.save!
           flash[:notice] = t("accept.success", scope: "decidim.admin.admin_terms_of_service")
-          redirect_to decidim_admin.root_path
+          redirect_to session[:user_return_to] || decidim_admin.root_path
```

Comment:
I think a better strategy here would be to call `stored_location_for` because it will wipe out the previous value from the session once it's used. The current strategy of fetching the value directly from the session would leave it there.
```suggestion
          redirect_to stored_location_for(current_user) || decidim_admin.root_path
```